vault/token_store.go

```diff
@@ -812,7 +839,7 @@ func (ts *TokenStore) store(ctx context.Context, entry *TokenEntry) error {
 // storeCommon handles the actual storage of an entry, possibly generating
 // secondary indexes
 func (ts *TokenStore) storeCommon(ctx context.Context, entry *TokenEntry, writeSecondary bool) error {
-	saltedID, err := ts.SaltID(ctx, entry.ID)
+	idHMAC, err := ts.hmac(ctx, entry.ID)
```

Comment:
This should look at the Version and do the right thing. If we storeCommon on a token that already existed, we'll now be saving it wth a new value but the old value will still be around.

Reply:
This is a great catch. Fixed.